app/src/main/java/io/apicurio/registry/ui/URLUtil.java

```diff
@@ -0,0 +1,103 @@
+package io.apicurio.registry.ui;
+
+import io.apicurio.common.apps.config.Info;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * Utility to generate absolute URLs.
+ *
+ * @author Jakub Senko <em>m@jsenko.net</em>
+ */
+@ApplicationScoped
+public class URLUtil {
+
+    @ConfigProperty(name = "registry.url.override.host")
+    @Info(category = "redirects", description = "Override the hostname used for generating externally-accessible URLs. " +
+            "The host and port overrides are useful e.g. when deploying with HTTPS passthrough ingress/route on OpenShift. " +
+            "In case like these, the request URL/IP (and port) that is then re-used for redirection " +
+            "does not belong to actual external URL used by the client, because the request is proxied. " +
+            "The redirection than fails because the target URL is not reachable.", availableSince = "2.5.0.Final")
+    Optional<String> urlOverrideHost;
+
+    @ConfigProperty(name = "registry.url.override.port")
+    @Info(category = "redirects", description = "Override the port used for generating externally-accessible URLs.", availableSince = "2.5.0.Final")
+    Optional<Integer> urlOverridePort;
+
+    @ConfigProperty(name = "registry.url.override.port.remove")
```

Comment:
Do we lose any semantics by removing this property and relying on the presence/absence of `registry.url.override.port` ?
Feels strange to have to set something, and have to set something more to remove it 😆

Reply:
You can have a k8s Service exposing HTTP on port 8123 for some reason. Then with default configuration we would remove the port and cause issues. The `registry.url.override.port.remove` property is basically a convenience to avoid setting `registry.url.override.port` to `443` or `80`.

But I can remove the  `registry.url.override.port.remove` functionality completely if it's confusing.

Reply:
Or we can autoremove 443 and 80 ports iff the protocol matches.

Reply:
I'll remove the property to make it less confusing